File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
res = 0
def dfs(cnt, cur, target, n, numbers):
    global res
    if cnt == n-1:
        if cur == target:
            res += 1
        return
    
    dfs(cnt+1, cur+numbers[cnt+1], target, n, numbers)
    dfs(cnt+1, cur-numbers[cnt+1], target, n, numbers)
    
            
def solution(numbers, target):
    global res
    answer = 0
    
    dfs(0, numbers[0], target, len(numbers), numbers)
    dfs(0, -numbers[0], target, len(numbers), numbers)
        
    return res

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;

    const stack = [[0, 0]]; // [현재합, 인덱스]

    while (stack.length) {
        const [cur, cnt] = stack.pop();

        if (cnt === numbers.length) {
            if (cur === target) {
                answer++;
            }
            continue;
        }

        stack.push([cur - numbers[cnt], cnt + 1]);
        stack.push([cur + numbers[cnt], cnt + 1]);
    }

    return answer;
}